cint/cint.py

```diff
@@ -94,18 +107,24 @@ def __rpow__(self, other):
         return self.__stronger_type(other)(calc(other) ** self.value)
 
     def __truediv__(self, other):
+        if self.__class__ in FLOATS or other in FLOATS:
+            return self.__stronger_type(other)(self.value / calc(other))
         return self.__stronger_type(other)(self.value // calc(other))
 
     # __div__ is Python 2 only
     __div__ = __floordiv__ = __truediv__
 
     def __rtruediv__(self, other):
+        if self in FLOATS or other in FLOATS:
+            return self.__stronger_type(other)(self.value / calc(other))
         return self.__stronger_type(other)(calc(other) // self.value)
 
     # __rdiv__ is Python 2 only
     __rdiv__ = __rfloordiv__ = __rtruediv__
 
     def __mod__(self, other):
+        if isinstance(other, (float, F32, F64)) or isinstance(self.value, (float, F32, F64)):
```

Comment:
If we had a `CFloat` type, then we could just keep this unimplemented ourselves?